safe/utilities/direction_distance.py

```diff
@@ -0,0 +1,84 @@
+# coding=utf-8
+
+"""Direction and Distance Functionality."""
+
+import logging
+from qgis.core import QgsDistanceArea
+
+LOGGER = logging.getLogger('InaSAFE')
+
+
+def bearing_to_cardinal(angle):
+    """Get cardinality of an angle.
+
+    :param angle: Bearing angle
+    :return:
+    """
+    direction_list = [
+        'N', 'NNE', 'NE', 'ENE', 'E', 'ESE', 'SE',
+        'SSE', 'S', 'SSW', 'SW', 'WSW', 'W',
+        'WNW', 'NW', 'NNW'
+    ]
+
+    bearing = float(angle)
+    direction_count = len(direction_list)
+    direction_interval = 360. / direction_count
+    index = int(round(bearing / direction_interval))
```

Comment:
Perhaps you can put reason why do you use round here. Not using `ceiling` or `floor`. And make sure you have unit test for any function that use `round`. It will help someone in the future who will migrate to python 3, because the behaviour is little bit different.
```python
# python 2.x
In [7]: [round(x) for x in [0.1, 0.5, 0.7, 1.1, 1.5, 1.7, 2, 2.1, 2.5]]
Out[7]: [0.0, 1.0, 1.0, 1.0, 2.0, 2.0, 2.0, 2.0, 3.0]

# python 3
>>> [round(x) for x in [0.1, 0.5, 0.7, 1.1, 1.5, 1.7, 2, 2.1, 2.5]]
[0, 0, 1, 1, 2, 2, 2, 2, 2]
```

Reply:
That code is the same code that realtime use in shake_event.py
https://github.com/inasafe/inasafe-realtime/blob/10f34187a3ace62817173537288ff675a6806766/src/realtime/earthquake/shake_event.py#L1635

I am not particularly sure if there is any particular reason in using rounding. I will try changing it